### ISSUE TITLE
Registration Tokens ... metadata leak?

### ISSUE THREAD

**Issue by willish** [vuvuzela/vuvuzela#30]
This may show a naive understanding of your work. So please forgive me if it is a silly question and a non-issue.

I was just curious about the necessity of acquiring a registration token...Surely the server providing this token can trivially link IP addresses with vuvuzela usernames?

**Comment by zeldovich** [comment]
The metadata that Vuvuzela tries to protect is "which users are communicating", rather than "which users are connected to Vuvuzela" or "what is this user's Vuvuzela username".

**Comment by willish** [comment]
Fair-point. I realised it was not an aim to protect "which users are connected", hence the argument for running the software all the time.

I do wonder if hiding the real identity of username X might be useful however. Even if an adversary cannot tell who "Snowden" is communicating with, having his IP address would be pretty useful, but I guess falls outside of the scope of Vuvuzela.

**Comment by zeldovich** [comment]
Indeed, I agree it would be useful in some cases to provide pseudonymity.

It might be interesting for us to think about how to provide pseudonymity.  One place where the current design doesn't provide pseudonymity is the add-friend protocol.  A user continuously polls the PKG servers for the user's add-friend key, so the PKG servers can determine the user's identity by correlating when the user goes online/offline and which keys start/stop being polled for.  Maybe we could do some variant of signature blinding to prevent the PKG server from learning the user's username..

**Comment by oltdaniel** [comment]
I would think a new issue would be a better place to specify the problem and a possible solution.
